File: Arrays _101/905. Sort Array By Parity/sortbyparity.py
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        base_idx = len(nums) - 1
        i = 0
        while i < base_idx:
            if nums[i] % 2 != 0:
                while nums[base_idx] % 2 != 0 and base_idx > i:
                    base_idx -= 1
                temp = nums[i]
                nums[i] = nums[base_idx]
                nums[base_idx] = temp
                base_idx -= 1
            i += 1
        return nums

File: 1. Arrays 101/905. Sort Array By Parity/sortbyparity.cpp
class Solution
{
public:
    vector<int> sortArrayByParity(vector<int> &nums)
    {
        int base_idx = nums.size() - 1;
        int temp;

        for (int i = 0; i < base_idx; i++)
        {
            if (nums[i] % 2 != 0)
            {
                while (nums[base_idx] % 2 != 0 && base_idx > i)
                {
                    base_idx--;
                }
                temp = nums[i];
                nums[i] = nums[base_idx];
                nums[base_idx] = temp;
                base_idx--;
            }
        }
        return nums;
    }
};